Master to trunk: primary renames (for rename day)

coder-karen opened the pull request:
## Not to be merged until AFTER the default branches are renamed to trunk

#### Changes proposed in this Pull Request:

* In preparation for renaming 'master' to 'trunk' in the Jetpack monorepo and mirror repos, this PR includes master to trunk renames in files that will need renaming directly after the default branch rename in GitHub. 
* The renames include files in:
   * .github/files: `lint-project-structure.sh`, `setup-wordpress-env.sh`, `pr-update-to.sh`, `renovate-close-old-PRs.sh`, `test-plugin-update/test.sh`, `test-plugin-update/prepare-zips.sh`
   * .github/workflows: `build-docker.yml`, `build.yml`, `codeql-analysis.yml`, `e2e-tests.yml`, `gardening.yml`, `linting.yml`, `phpcompatibility-dev.yml` `pr-is-up-to-date.yml`, `renovate-cleanup.yml`, `slack-workflow-failed.yml`, `tests.yml`
   * projects/github-actions (all in this group changed from 'master' to 'main': `pr-is-up-to-date/tests/funcs.sh`, `pr-is-up-to-date/tests/test-010-init_repo.sh`, `pr-is-up-to-date/tests/test-020-fetch_prs.sh`, `pr-is-up-to-date/tests/test-030-should_process_pr.sh`, `pr-is-up-to-date/tests/test-040-process_pr.sh`, `repo-gardening/src/tasks/add-milestone/index.js`, `repo-gardening/src/tasks/wpcom-commit-reminder/index.js`
   * projects/js-packages: `eslint-changed/tests/cli.test.js`
   * projects/packages: `analyzer/scripts/example.php`, `analyzer/scripts/jetpack-example.php`, `analyzer/src/api/class-plugin-downloader.php`, `changelogger/tests/php/tests/src/AddCommandTest.php`
   * tools: `cli/commands/changelog.js`,  `cli/commands/build.js`, `cli/commands/generate.js`, `create-release-branch.sh`, `jetpack-live-branches/jetpack-live-branches.user.js`, `test-on-wpcom`, `check-intra-monorepo-deps.sh`, `composer-update-monorepo.sh`, `js-tools/git-hooks/pre-push-hook.js`
   * tools/e2e-commons: `bin/tests/resources/config.json`, `bin/tests/resources/gh-context-pr.json`, `bin/tests/resources/gh-context-push.json`, `config/default.cjs`

Not all of these are necessarily breaking, but could impact tests / other actions that may run on new PRs or pushes. Also there are some occasional comment changes where files were already being edited as well.


**Files not yet added:**

*  `composer.json` and `composer.lock` files and other composer related files such as `installed.json`, `installed.php`(with `dev-master` wording renamed) to be added just before PR merge. A script will be run to rename relevant files (steps outlined in Asana).




#### Jetpack product discussion

p9dueE-59T-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Proof-read - all you should see is changes from master to trunk, nothing different added.
* If any name changes show up that you believe can be made ahead of time (or aren't needed), please do flag that.
* This PR is not for merging. It should be the first PR to be merged after both the mirror repos and the main Jetpack monorepo have had their default branches renamed.

coder-karen committed:
First .github files with master-trunk renames

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

coder-karen committed:
[not verified] js-packages and packages files with master to trunk renames

coder-karen committed:
changelog

coder-karen committed:
[not verified] tools files with master to trunk renames

coder-karen committed:
monorepo root files with master to trunk renames

coder-karen committed:
[not verified] projects/github-actions files with master to trunk renames

coder-karen committed:
Adding GitHub action changelog files

coder-karen committed:
Adding .github/workflows files with master to trunk renames

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

anomiex reviewed (commented):
> Files still to be added with edits:
>
> `tools/js-tools/git-hooks/pre-push-hook.js`

Yes, that will definitely be needed.

> Files not added:
>
> `.github/files/test-plugin-update/prepare-zips.sh`

As noted inline, that will be needed.

> Any files referencing `dev-master`, eg. composer.json and lock files primarily (the current thought is to have these in a separate PR but this needs more investigation).

As noted in p1654790703815559-slack-C03FX4HLD9B, that will probably be needed after all.

----

Additionally,

* These two lines should probably be updated to avoid confusing people using the CLI: https://github.com/Automattic/jetpack/blob/9c71427f40cfc5b4b400996094d11915a48920a1/tools/cli/commands/release.js#L139 https://github.com/Automattic/jetpack/blob/9c71427f40cfc5b4b400996094d11915a48920a1/tools/cli/commands/release.js#L148
* Speaking of which, other references to "master" in the docs (outside of URLs) should probably also be prepared for release-day too. The ones in URLs aren't so important, AFAIK they'll just redirect.

Also, BTW, I noticed that we forgot to update this in #24083. https://github.com/Automattic/jetpack/blob/9c71427f40cfc5b4b400996094d11915a48920a1/projects/plugins/jetpack/tests/action-test-php.sh#L9 So we haven't been running the multisite tests since then. Fixing that in #24690. I wonder how many such things this change is going to cause.

coder-karen commented on .eslines.json:
```diff
@@ -5,7 +5,7 @@
 	"processors": {
 		"downgrade-unmodified-lines": {
-			"remote": "origin/master",
+			"remote": "origin/trunk",
```
> We may as well just delete the whole thing.

Sounds good.

coder-karen commented on .github/files/test-plugin-update/test.sh:
```diff
@@ -13,7 +13,7 @@ done
 cd /var/www/html
 for SLUG in "${SLUGS[@]}"; do
-	for FROM in stable master dev; do
+	for FROM in stable trunk dev; do
```
Sounds good. That (prepare-zips.sh) file was something I wanted to check in about, as a result of the Slack conversation you linked too. So this helps, I'll make changes to that file too.

coder-karen replied:
> OTOH, line 29 should either not be updated or should be updated to something like

I will make that suggested change as it seems that it allows more flexibility (potentially one less thing to change later if/when we follow up on the suggestions in that Slack thread).

coder-karen commented on projects/github-actions/pr-is-up-to-date/changelog/update-monorepo-master-refs-primary:
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: changed
+
+Renaming master references to trunk in GH action files, to work with the GitHub default branch name changing to trunk.
```
Thanks for explaining this! Since we're going through everything, I'll change it all to main (except in those two repo-gardening files). I'll commit your suggestion here too.

coder-karen committed:
Update projects/github-actions/pr-is-up-to-date/changelog/update-monorepo-master-refs-primary

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

coder-karen committed:
Update projects/js-packages/eslint-changed/changelog/update-monorepo-master-refs-primary

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

coder-karen committed:
Update projects/packages/changelogger/changelog/update-monorepo-master-refs-primary

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

coder-karen committed:
Changing some refs to main where appropriate and removing .eslines.json

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

coder-karen committed:
Merge remote-tracking branch 'origin/update/monorepo-master-refs-primary' into update/monorepo-master-refs-primary

coder-karen commented:
Thanks for your reviews @anomiex ! I've added new commits with the suggestions. I'll hold out with pre-push-hooks.js just a little longer as it's useful just now :) 

> Speaking of which, other references to "master" in the docs (outside of URLs) should probably also be prepared for release-day too. The ones in URLs aren't so important, AFAIK they'll just redirect.

Yes, this is the next PR to work on (after the master and dev-master changes). That includes things like comments and CLI refs too - anything 'non disruptive' really. It may make sense to update URLs in some cases (eg. the more obvious MD files related to developing in the monorepo), but it will redirect as you thought, so I'll see how I get on with this. Following that (so after the rename), I'll be going through other internal documentation (including the Field Guide) - if possible I'll start writing up a list of the most obvious docs to update ahead of time and adding that to Asana too.

As a next step just now I'll start looking at dev-master renames (outside of composer.json and .lock files).

What this should mean though is when almost complete, this PR has all master and dev-master rewording except for relevant composer.json and composer.lock files. This would happen on the day - I've listed some steps in Asana on what I'll do there - and get added to this PR. I've updated this PR description to cover this too.

anomiex commented on projects/github-actions/pr-is-up-to-date/changelog/update-monorepo-master-refs-primary:
```diff
@@ -1,4 +1,4 @@
 Type: changed
 
-Renaming master references to trunk in GH action files, to work with the GitHub default branch name changing to trunk.
+Renaming `master` to `trunk` in tests.
```
Well now you made it "main" 😀
```suggestion
Renaming `master` to `main` in tests.
```

coder-karen replied:
😬

anomiex commented:
> (outside of composer.json and .lock files)

Note once you update the composer.json files, then `tools/fixup-project-versions.sh` should handle the lock files for you.

anomiex commented on projects/github-actions/pr-is-up-to-date/tests/test-010-init_repo.sh:
```diff
@@ -13,7 +13,7 @@ test_make_merge E D1 D2
 test_make_commits F G H I
 test_make_commits '' O1 O2 O3
-test_make_branch master I
+test_make_branh main I
```
Typo here
```suggestion
test_make_branch main I
```

coder-karen replied:
Thank you.

coder-karen committed:
dev-master wording renames

coder-karen committed:
Update projects/github-actions/pr-is-up-to-date/changelog/update-monorepo-master-refs-primary

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

coder-karen committed:
Update projects/github-actions/pr-is-up-to-date/tests/test-010-init_repo.sh

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

coder-karen committed:
Typo fix and correcting changelog entry

coder-karen committed:
Updating changelog in projects/github-actions/pr-is-up-to-date

coder-karen committed:
dev-master wording renames

coder-karen committed:
changelog

coder-karen committed:
Merge remote-tracking branch 'origin/update/monorepo-master-refs-primary' into update/monorepo-master-refs-primary

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

coder-karen committed:
Adding lint-project-structure.sh to files with dev-master rewording

anomiex reviewed (commented):
Looks good to me, other than the things noted (`composer.json` aliases, `tools/js-tools/git-hooks/pre-push-hook.js`).

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

coder-karen committed:
Changing master ref to trunk in pre-push-hook.js

coder-karen commented:
The `pre-push-hook.js` file is added now as it is the last non-composer-related file, so the last files - relevant `composer.json and `composer.lock` files - will be added on rename day.

coder-karen committed:
Changing master to trunk in a few comments in .github/workflows/linting.yml

adimoldovan commented:
@coder-karen there is one more reference to master in `tools/e2e-commons/config/encrypted.enc`. You'll need to decrypt the file first and encrypt it again before pushing. `config:decrypt` and `config:encrypt` pnpm scripts will help. See [this readme](https://github.com/Automattic/jetpack/tree/master/projects/plugins/jetpack/tests/e2e#test-configuration) for more and let me know if I can help with anything.

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

coder-karen committed:
Adding master to trunk wording change in encrypted.enc

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

coder-karen committed:
Merge remote-tracking branch 'origin/master' into update/monorepo-master-refs-primary

coder-karen committed:
Renaming master references in composer files to trunk
